### PR TITLE
Integration Event Message to EventHub

### PR DESCRIPTION
**Pull request by Korgath**
## Description

- Timertrigger function added: EventMessageDispatcher
- Protobuff contract implemented including a dispatcher and channel
- Simpleinjector implemented
- When EF is fully implemented a repository fetching from the Outbox will be added instead of hardcoded object

## References

* #44

### CODE REVIEW

File: source/Energinet.DataHub.MeteringPoints.EntryPoints.Outbox/EventMessageDispatcher.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            _dispatcher = dispatcher;
+        }
+
+        [Function("EventMessageDispatcher")]
+        public async Task RunAsync(
+        [TimerTrigger("%EVENT_MESSAGE_DISPATCH_TRIGGER_TIMER%")] string timerInformation,
+        FunctionContext context)
+        {
+            var logger = context.GetLogger("EventMessageDispatcher");
+            logger.LogInformation($"C# Timer trigger function executed at: {DateTimeOffset.UtcNow} (UTC)");
```
Comment (erduna) [inline source/Energinet.DataHub.MeteringPoints.EntryPoints.Outbox/EventMessageDispatcher.cs:39]: Maybe some more telling info?

Comment (erduna) [reply source/Energinet.DataHub.MeteringPoints.EntryPoints.Outbox/EventMessageDispatcher.cs:39]: Maybe the dispatched event?

File: source/Energinet.DataHub.MeteringPoints.EntryPoints.Processing/CreateMeteringPointHttpTrigger.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Http;
+using Microsoft.Extensions.Logging;
+
+namespace Energinet.DataHub.MeteringPoints.EntryPoints.Processing
+{
+    public class HttpTrigger
+    {
+        [Function("CreateMeteringPoint")]
+        public async Task<HttpResponseData> RunAsync(
```
Comment (cmgrd) [inline source/Energinet.DataHub.MeteringPoints.EntryPoints.Processing/CreateMeteringPointHttpTrigger.cs:31]: Should this be here?

File: source/Energinet.DataHub.MeteringPoints.Infrastructure/IntegrationServices/Mappers/IntegrationEventMessageMapper.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    public class IntegrationEventMessageMapper : ProtobufOutboundMapper<IntegrationEventMessage>
+    {
+        protected override IMessage Convert(IntegrationEventMessage obj)
+        {
+            if (obj == null) throw new ArgumentNullException(nameof(obj));
+            return new IntegrationEventEnvelope()
+            {
+                IntegrationEventMessage = new Contracts.IntegrationEventMessage
+                {
+                    Id = obj.Id, Type = obj.Type, Data = obj.Data,
```
Comment (cmgrd) [inline source/Energinet.DataHub.MeteringPoints.Infrastructure/IntegrationServices/Mappers/IntegrationEventMessageMapper.cs:31]: Nit: I'd prefer newlines.

File: source/Energinet.DataHub.MeteringPoints.Infrastructure/IntegrationServices/IntegrationEventMessage.cs [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using Energinet.DataHub.MeteringPoints.Application.Transport;
+using MediatR;
+
+namespace Energinet.DataHub.MeteringPoints.Infrastructure.IntegrationServices
+{
+    public class IntegrationEventMessage : IOutboundMessage, IInboundMessage
```
Comment (cmgrd) [inline source/Energinet.DataHub.MeteringPoints.Infrastructure/IntegrationServices/IntegrationEventMessage.cs:21]: Why inbound?

File: source/Energinet.DataHub.MeteringPoints.Infrastructure/IntegrationServices/IntegrationEventMessage.cs [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    {
+        public string? Id { get; set; }
+
+        public string? Type { get; set; }
+
+        public string? Data { get; set; }
```
Comment (cmgrd) [inline source/Energinet.DataHub.MeteringPoints.Infrastructure/IntegrationServices/IntegrationEventMessage.cs:27]: Nullables?

File: source/Energinet.DataHub.MeteringPoints.EntryPoints.Outbox/EventMessageDispatcher.cs [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+            logger.LogInformation($"From function timer trigger input: {timerInformation}");
+
+            // TODO: Change this to use a repository when EF is implemented
```
Comment (cmgrd) [inline source/Energinet.DataHub.MeteringPoints.EntryPoints.Outbox/EventMessageDispatcher.cs:42]: You could consider introducing an application service to orchestrate these.

**Review by cmgrd — CHANGES_REQUESTED** [review]

File: source/Energinet.DataHub.MeteringPoints.EntryPoints.Outbox/EventMessageDispatcher.cs (hunk 6 above)
Comment (Korgath) [reply source/Energinet.DataHub.MeteringPoints.EntryPoints.Outbox/EventMessageDispatcher.cs:42]: Yes, this was the plan. Will update TODO to reflect this

File: source/Energinet.DataHub.MeteringPoints.Infrastructure/IntegrationContracts/Contract.proto [hunk 7]
```diff
@@ -38,4 +38,16 @@ message MeteringPointEnvelope {
   oneof MeteringPointMessages {
     CreateMeteringPoint createMeteringPoint = 1;
   }
+}
+
+message IntegrationEventMessage {
+  string id = 1;
+  string type = 2;
+  string data = 3;
```
Comment (cmgrd) [inline source/Energinet.DataHub.MeteringPoints.Infrastructure/IntegrationContracts/Contract.proto:46]: Where are these defined?

File: source/Energinet.DataHub.MeteringPoints.Infrastructure/IntegrationContracts/Contract.proto [hunk 8]
```diff
@@ -38,4 +38,16 @@ message MeteringPointEnvelope {
   oneof MeteringPointMessages {
     CreateMeteringPoint createMeteringPoint = 1;
   }
+}
+
+message IntegrationEventMessage {
```
Comment (cmgrd) [inline source/Energinet.DataHub.MeteringPoints.Infrastructure/IntegrationContracts/Contract.proto:43]: I think this should be a new proto file as this is another contract

**Review by cmgrd — CHANGES_REQUESTED** [review]

File: source/Energinet.DataHub.MeteringPoints.Infrastructure/IntegrationContracts/Contract.proto (hunk 7 above)
Comment (Korgath) [reply source/Energinet.DataHub.MeteringPoints.Infrastructure/IntegrationContracts/Contract.proto]: In the Outbox table

File: source/Energinet.DataHub.MeteringPoints.EntryPoints.Outbox/EventMessageDispatcher.cs (hunk 1 above)
Comment (Korgath) [reply source/Energinet.DataHub.MeteringPoints.EntryPoints.Outbox/EventMessageDispatcher.cs:39]: i'll update the logging info, when doing the application service which will be orchestrating this dispatcher

File: source/Energinet.DataHub.MeteringPoints.Infrastructure/IntegrationContracts/IntegrationEventContract.proto [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+
+syntax = "proto3";
+
+// import "google/protobuf/timestamp.proto";
+
+option csharp_namespace = "Energinet.DataHub.MeteringPoints.IntegrationEventContracts";
+
+message IntegrationEventMessage {
+  string id = 1;
+  string type = 2;
```
Comment (cmgrd) [inline source/Energinet.DataHub.MeteringPoints.Infrastructure/IntegrationContracts/IntegrationEventContract.proto:24]: If defined in the database table how does the client of the contract know about them?

**Review by cmgrd — CHANGES_REQUESTED** [review]

**Pull request closed without merging**